os/kernel/binary_manager/binary_manager.h

```diff
@@ -239,9 +237,9 @@ binmgr_uinfo_t *binary_manager_get_udata(uint32_t bin_idx);
 #define BIN_OFFSET(bin_idx)                             binary_manager_get_udata(bin_idx)->load_attr.offset
 #define BIN_STACKSIZE(bin_idx)                          binary_manager_get_udata(bin_idx)->load_attr.stack_size
 #define BIN_PRIORITY(bin_idx)                           binary_manager_get_udata(bin_idx)->load_attr.priority
-#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
 #define BIN_LOADINFO(bin_idx)                           binary_manager_get_udata(bin_idx)->binp
-#endif
+#define BIN_BINARY_HEAP_PTR(bin_idx)                    binary_manager_get_udata(bin_idx)->binp->uheap
```

Comment:
1. Could you let me know why you need `BIN_LOADINFO` for this? I mean why do you remove the conditional?
2. When we call malloc at the kernel, it gets proper app heap and allocates new memory there. Do we need this newly?

Reply:
1. `BIN_LOADINFO` is needed for `BIN_BINARY_HEAP_PTR`. It is required for getting app heap pointer in kernel. 
2. In kernel, if we call malloc, it gets the CURRENT app heap using the currently active tcb (tcb->uheap). But, in this case, we want to get heap pointer of other app. So, it cannot be done using the tcb. We need to get it from the other app binary loading data struct.